Provide structure through the URL parameter, implement mof checkers.

yakutovicha opened the pull request.

yakutovicha committed:
Provide structure through the URL parameter.

yakutovicha committed:
Get URL to the structure from the URL parameter.

kjappelbaum commented on multistage_geo_opt_ddec.ipynb:
```diff
@@ -124,6 +125,22 @@
+    "try:\n",
+    "    structure_url = urlparse.parse_qs(url.query)['structure'][0]\n",
+    "    from ase.io import read\n",
+    "    structure_widget.structure = read(io.StringIO(structure_url), format='cif')\n",
```
one thing we might want to think about is if we want to leave if hard coded at `CIF` or also allow `xyz`, `mol`, `pdb` etc. We could provide the format as additional url parameter.

yakutovicha replied:
It is done by the eln importer, this guy is really outdated.

kjappelbaum commented:
@yakutovicha can you maybe push the most recent version you have?

yakutovicha committed:
compute isotherm notebook: allow to provide the structure uuid through url parameters.

yakutovicha committed:
pore analysis notebook: allow to provide the structure uuid through url parameters.

yakutovicha committed:
Add mof checker tool

yakutovicha committed:
Multistage notebook:

1. Provide the structure through the URL params.
2. Add mof checker and editor.

yakutovicha committed:
MofChecker: only run selected checks.

yakutovicha committed:
Improve MofCheckerWidget

kjappelbaum commented on utils/mof_checker/__init__.py:
```diff
@@ -0,0 +1,123 @@
+        self.mfchk = MOFChecker.from_ase(self.structure, primitive=False)
+        self.checks = []
+        if self.structure:
+            self.checks = [ check for (check_name, check) in self.mfchk.checks.items() if check_name in ENABLED_CHECKS]
```
```suggestion
            self.checks = [check for (check_name, check) in self.mfchk.checks.items() if check_name in ENABLED_CHECKS]
```

kjappelbaum commented on utils/mof_checker/__init__.py:
```diff
@@ -0,0 +1,123 @@
+                        selection = check.flagged_indices
+                    button = ipw.Button(description="Select", layout={"width": "initial"})
+                    button.on_click(fct.partial(self._select_atoms, selection=selection))
+                    #text = ipw.HTML("Found " + check.name.lower() + ": " + ', '.join(map(str, selection)))
```
```suggestion
```

kjappelbaum commented on utils/mof_checker/__init__.py:
```diff
@@ -0,0 +1,123 @@
+        atoms = self.structure.copy()
+        selection = self.selection
+
+        h_positions = self.mfchk.undercoordinated_c_candidate_positions + self.mfchk.undercoordinated_n_candidate_positions
```
I'll double check if the list is always flat. I think this is atm not good in `mofchecker` as it might return you a list of lists of positions in case you have multiple kinds (sp2, sp3...) missing H

yakutovicha replied:
>  it might return you a list of lists of positions in case you have multiple kinds (sp2, sp3...) missing H

eh, maybe this level of detail isn't really needed.

kjappelbaum commented on utils/mof_checker/__init__.py:
```diff
@@ -0,0 +1,123 @@
+
+    def __init__(self, **kwargs):
+        self.text = ipw.HTML()
+        self.check_button = ipw.Button(description="Final Structure Check", button_style = 'warning')
```
```suggestion
        self.check_button = ipw.Button(description="Final Structure Check", button_style='warning')
```

kjappelbaum commented on utils/mof_checker/__init__.py:
```diff
@@ -0,0 +1,123 @@
+                    button = ipw.Button(description="Select", layout={"width": "initial"})
+                    button.on_click(fct.partial(self._select_atoms, selection=selection))
+                    #text = ipw.HTML("Found " + check.name.lower() + ": " + ', '.join(map(str, selection)))
+                    text = ipw.HTML("Found " + check.name.lower())
```
if you have suggestions for better names, let me know I'm happy to change them!

yakutovicha replied:
Not at the moment, but I know whom to contact in case I am not happy with a name :D

kjappelbaum reviewed (dismissed):
Many thanks, lgtm!

yakutovicha committed:
Apply suggestions from code review

Co-authored-by: Kevin Jablonka <32935233+kjappelbaum@users.noreply.github.com>
